File: python-fastapi/app/service.py
from .errors import ServiceError
from .models import Event, EventInput
from .store import MemoryStore


class EventService:
    def __init__(self, store: MemoryStore):
        self.store = store

    def create(self, data: EventInput) -> Event:
        raise NotImplementedError

    def read(self, event_id: str) -> Event:
        raise NotImplementedError

    def update(self, event_id: str, data: EventInput) -> Event:
        raise NotImplementedError

    def delete(self, event_id: str) -> None:
        raise NotImplementedError

File: javascript-express/src/service.js
import { NotImplementedError, ServiceError } from "./errors.js";
import { Event, EventInput } from "./models.js";

export class EventService {
  constructor(store) {
    this.store = store;
  }

  create(data) {
    throw new NotImplementedError();
  }

  read(eventId) {
    throw new NotImplementedError();
  }

  update(eventId, data) {
    throw new NotImplementedError();
  }

  delete(eventId) {
    throw new NotImplementedError();
  }
}
